src/unity/Assets/Crest/Scripts/Shapes/ShapeGerstnerBatched.cs

```diff
@@ -348,6 +364,92 @@ public Vector3 GetDisplacement(ref Vector3 worldPos, float toff)
             return result;
         }
 
+        public bool m_Approx = true;
+
+        public float SinCos(float x,out float cos)
+        {
+//            if (!m_Approx)
+            {
+                cos = Mathf.Cos(x);
+                return Mathf.Sin(x);
+            }
+//            cos = Mathf.Cos(x);
+//            return Mathf.Sqrt(1 - cos * cos);
+
+
+            /*
+                        float xf = Mathf.Floor(x / (Mathf.PI * 2));
+                        x -= xf * Mathf.PI * 2;
+                        const float B = 4 / Mathf.PI;
+                        const float C = -4 / (Mathf.PI * Mathf.PI);
+
+                        float Sin= -(B * x + C * x * ((x < 0) ? -x : x));
+
+                        cos = Mathf.Sqrt(1 - Sin * Sin);
+
+                        return Sin;
+            */
+        }
+
+        Dictionary<uint,Vector3> m_Recent=new Dictionary<uint, Vector3>();
+
+        uint CalcHash(Vector3 _wp)
```

Comment:
great approach with the hashing :).

the small idea occurred to me, could we do this with int instead of uint? that might allow this to reduce to:
return (int)(_wp.x + _wp.z*65000);

fairly trivial i guess but if calling this many times per frame it could make a small difference.

Reply:
thinking about it I think you need to make one axis positive. because -1,1 would be the same as 64999,0